nanoFramework.Logging.Syslog/SyslogLogger.cs

```diff
@@ -0,0 +1,76 @@
+﻿using System;
+using System.Reflection;
+using Microsoft.Extensions.Logging;
+
+namespace nanoFramework.Logging.Syslog
+{
+    /// <summary>
+    /// ILogger implmentation that send message log to a Syslog server following RFC3164 convention.
+    /// While not mandatory it is recommended to use <see cref="SyslogLoggerFactory"/> to create the SyslogLogger instances.
+    /// </summary>
+    public class SyslogLogger : ILogger
+    {
+        private SyslogClient _client;
+
+        /// <summary>
+        /// Create new logger based on existing SyslogClient. Recommended to use LoggerFactory patern trough SyslogLoggerFactory instead of direct use.
+        /// </summary>
+        /// <param name="client">SyslogClient to use for sending message</param>
+        /// <param name="categoryName">CategoryName of this logger used as TAG in the Syslog messages</param>
+        /// <exception cref="ArgumentNullException"></exception>
+        public SyslogLogger(SyslogClient client, string categoryName)
+        {
+            if (client is null)
+                throw new ArgumentNullException(nameof(client));
+            
+            _client = client;
```

Comment:
remove the previous if and replace by:
```suggestion
            _client = client ?? throw new ArgumentNullException(nameof(client));
```

Reply:
If there are other places, I'll just make a short version